ethash/src/cache.rs

```diff
@@ -185,7 +186,12 @@ fn make_memmapped_cache(path: &Path, num_nodes: usize, ident: &H256) -> io::Resu
 
 	let mut memmap = unsafe { MmapMut::map_mut(&file)? };
 
-	unsafe { initialize_memory(memmap.as_mut_ptr() as *mut Node, num_nodes, ident) };
+	unsafe {
+		let bytes = memmap.as_mut_ptr();
+		assert_eq!(bytes as usize % mem::align_of::<Node>(), 0, "Mmap memory not aligned.");
```

Comment:
what's the rationale of panicking instead of returning an Error as in the code below?

Reply:
is there any way to check this at compile-time? 

[mem::align_of](https://doc.rust-lang.org/std/mem/fn.align_of.html) is const fn